src/iam/certhandler/certmodule.cpp

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "aos/iam/certmodule.hpp"
+#include "log.hpp"
+
+namespace aos {
+namespace iam {
+namespace certhandler {
+
+/***********************************************************************************************************************
+ * Public CertModule methods
+ **********************************************************************************************************************/
+
+CertModule::CertModule(const String& certType, const ModuleConfig& config)
+    : mCertType(certType)
+    , mModuleConfig(config)
+{
+}
+
+Error CertModule::Init(crypto::x509::ProviderItf& x509Provider, HSMItf& hsm)
+{
+    mX509Provider = &x509Provider;
+    mHSM = &hsm;
+
+    if (mModuleConfig.mSkipValidation) {
+        LOG_WRN() << "Skip validation for certType: " << GetCertType();
+
+        return Error::Enum::eNone;
+    }
+
+    return mHSM->ValidateCertificates(mValidCerts, mInvalidCerts, mInvalidKeys);
+}
+
+void CertModule::SetListener(CertListener& listener)
+{
+    mCertListener = &listener;
+}
+
+const String& CertModule::GetCertType() const
+{
+    return mCertType;
+}
+
+const Array<CertInfo>& CertModule::GetValidCerts() const
+{
+    return mValidCerts;
```

Comment:
We didn't keep it in memory just to reduce memory usage. We have a bunch of certificates probably up 15-20 for each node. It could be an option as well but in this case, it is useless to keep these certificated in storage at all. If we keep this info in memory, no need to read it each time from DB (disk).  `CertModule` should provide required info for `CertHandler`.